cdk/src/open-data-platform/data-plane/data-import/hander-factory.ts

```diff
@@ -0,0 +1,107 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { ConnectionPool } from '@databases/pg';
+import * as AWS from 'aws-sdk';
+import { connectToDb } from '../schema/schema.handler';
+
+// These libraries don't have types, so they are imported in a different way.
+const { chain } = require('stream-chain');
+const { streamArray } = require('stream-json/streamers/StreamArray');
+const Batch = require('stream-json/utils/Batch');
+const Pick = require('stream-json/filters/Pick');
+
+const S3 = new AWS.S3();
+
+/**
+ * Builds a lambda handler that parses a GeoJSON file and executes a callback.
+ * @param s3Params - Details for how to get the GeoJSON file.
+ * @param callback - Function that operates on a single element from the file.
+ * @param numberOfRowsToWrite - Row limit, which can be used for testing.
+ * @returns
+ */
+export const geoJsonHanderFactory = (
+  s3Params: AWS.S3.GetObjectRequest,
+  callback: (row: any, db: ConnectionPool) => Promise<void>,
+  numberOfRowsToWrite: number = Infinity,
+): ((event: APIGatewayProxyEvent) => Promise<APIGatewayProxyResult>) => {
+  /**
+   * Reads an GeoJSON file and performs the callback on each element. This does not handle any
+   * processing of the data itself, but provides a DB connection to the callback to process it.
+   *
+   * TODO: track how many rows were processed and avoid re-importing the old rows.
+   *
+   * @param db - Connection pool to the DB, made available to the callback.
+   * @returns
+   */
+  const readGeoJsonFile = (db: ConnectionPool): Promise<number> =>
```

Comment:
Mostly for my own knowledge: how come this is in the form of const name = (args) => Promise instead of async function name(args) {}?

Reply:
I find arrow function notation easier to read. But there's really any difference otherwise. It doesn't have the `async` keyword because its not a async function-- it returns a hand-written promise instead.

I actually used it in this PR because I needed to look up how to say a function returns a function in Typescript and it used arrow notation.